pkg/authenticator/config/config.go

```diff
@@ -28,88 +29,109 @@ const (
 	DefaultClientCertPath = "/etc/conjur/ssl/client.pem"
 	DefaultTokenFilePath  = "/run/conjur/access-token"
 
+	DefaultConjurVersion = "5"
+
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
 	DefaultTokenRefreshTimeout = 6 * time.Minute
 )
 
-// New returns a new authenticator configuration object
+var requiredEnvVariables = []string{
+	"CONJUR_AUTHN_URL",
+	"CONJUR_ACCOUNT",
+	"CONJUR_AUTHN_LOGIN",
+	"MY_POD_NAMESPACE",
+	"MY_POD_NAME",
+}
+
+// NewFromEnvFunc defines the method for creating a new config from the env
+type NewFromEnvFunc func(Options) (*Config, error)
+
+// ReadSSLCertFunc defines the method for reading an SSL Certificate from the env
+type ReadSSLCertFunc func() ([]byte, error)
```

Comment:
Instead of mocking the `readSSLCert` function, you could get better coverage by mocking the lower level file reader inside the `readSSLCert` function itself, i.e. this line:
```
return ioutil.ReadFile(SSLCertPath)
```
The way to do this would be to define a `fileReader` interface, and add a `fileReader` interface as an argument to the `FromEnv()` function signature (no need to use functional options since there's only one option, and it's a simple interface). Within `FromEnv()`, you would pass on that interface to the call to `readSSLCert()`, and inside `readSSLCert()`, you would do the reading of the file using the interface passed in.
Then in your UT, you would pass in a buffer or some other IO reader as a mock.